app/init.php

```diff
@@ -1796,6 +1811,27 @@ function getDevice(string $root, string $connection = ''): Device
     return [];
 });
 
+App::setResource('devKey', function ($request, $project, $dbForConsole) {
+    $devKey = $request->getHeader('x-appwrite-development-key', '');
+    // Check if given key match project's development keys
+    $key = $project->find('secret', $devKey, 'devKeys');
+    if ($key) {
+        $expire = $key->getAttribute('expire');
+        if (!empty($expire) && $expire < DatabaseDateTime::formatTz(DatabaseDateTime::now())) {
+            return new Document([]);
+        }
+
+        $accessedAt = $key->getAttribute('accessedAt', '');
+        if (DatabaseDateTime::formatTz(DatabaseDateTime::addSeconds(new \DateTime(), -APP_KEY_ACCESS)) > $accessedAt) {
+            $key->setAttribute('accessedAt', DatabaseDateTime::now());
+            Authorization::skip(fn () => $dbForConsole->updateDocument('keys', $key->getId(), $key));
+            $dbForConsole->purgeCachedDocument('projects', $project->getId());
+        }
```

Comment:
I think this will run on every request. I see `APP_KEY_ACCESS` which probably is meant to avoid that, but I dont think it does. It only shifts the point when it starts occuring

Reply:
What do you mean, here we have `addSeconds` and when we provide `-APP_KEY_ACCESS` it subtracts the value of `APP_KEY_ACCESS` from the current time.

Reply:
I cant remember my line of thought anymore.
Let's add var_dump inside the `if`, and do a few requests. See how often var_dump happens.

Ignore this comment if it doesnt happen on every request

Reply:
Yes, it doesn't